awx/main/tasks/jobs.py

```diff
@@ -566,6 +566,11 @@ def run(self, pk, **kwargs):
             status = res.status
             rc = res.rc
 
+            # We call this to get the current values from the database, in case update_model was called
+            # within the threadpools inside of AWXReceptorJob. We use update_model instead of
+            # refresh_from_db here because it contains retry logic that is resilient to database failures.
+            self.instance = self.update_model(self.instance.pk)
```

Comment:
I think this should replace the `self.instance = self.update_model(pk)` on line 599 / 604. You can just read the code, and know for absolute sure that does nothing helpful. It's called right before another `update_model` call, and the first thing _that_ does is get the object again. Swapping this for that will leave it more functional without adding more noise.

Reply:
I think I agree.